llvm/test/Transforms/LoopIdiom/memcpy-debugify-remarks.ll

```diff
@@ -10,12 +10,11 @@ target triple = "x86_64-unknown-linux-gnu"
 
 ; Check that everything still works when debuginfo is present, and that it is reasonably propagated.
 
-; CHECK: remark: <stdin>:6:1: Formed a call to llvm.memcpy.p0.p0.i64() intrinsic from load and store instruction in test6_dest_align function{{$}}
+; CHECK: remark: <unknown>:0:0: Formed a call to llvm.memcpy.p0.p0.i64() intrinsic from load and store instruction in test6_dest_align function{{$}}
```

Comment:
Dropping the location for `NewCall` is the right thing - but it seems to me that the optimization remark would be more useful if we retained the old debug loc, which is still correct for attribution purposes (though unfortunately we can't separate stepping and attribution concerns at the moment); you could change the optimization remark to refer to `TheStore`'s DebugLoc, and we could get a meaningful source location here.

Reply:
Ok, I've already changed the optimization remark to refer to `TheStore`'s DebugLoc and reverted the change in this  check.